js/style/style_layer.js

```diff
@@ -152,19 +156,29 @@ StyleLayer.prototype = util.inherit(Evented, {
         }
     },
 
-    getPaintValue: function(name, zoom, zoomHistory) {
+    getPaintValue: function(name, globalProperties, featureProperties) {
         var specification = this._paintSpecifications[name];
         var transition = this._paintTransitions[name];
 
         if (transition) {
-            return transition.at(zoom, zoomHistory);
+            return transition.calculate(globalProperties, featureProperties);
         } else if (specification.type === 'color' && specification.default) {
             return parseColor(specification.default);
         } else {
             return specification.default;
         }
     },
 
+    isPaintValueFeatureConstant: function(name) {
+        var transition = this._paintTransitions[name];
+
+        if (transition) {
+            return transition.declaration.isFeatureConstant;
```

Comment:
You're right. Good catch. I used to set `StyleDeclaration#isFeatureConstant` but that code was lost in a rebase. Pushing a fix